### ISSUE TITLE
File target doesn't duplicate header in archived files

### ISSUE THREAD

**Issue by johnnyjob** [NLog/NLog#245]
File target writes header only once. If archiving is configured it would be useful to see the header in all log files, including archives. Use this sample project to reproduce the bug: https://skydrive.live.com/redir?resid=DF959DE37F64E71D!157&authkey=!AHQsJ6pHwtx1Mbk

Steps to reproduce:
1. Open sample project, compile and run

Expected results:
- Header is included into all log files

Actual results:
- Header is included into the first file only (Log-6756-2013-07-23.0.txt).

Reproduced in NLog version 2.0.0.2 for .Net 3.5 (actual binaries are included into the sample project).

I suggest to make the following patch to FileTarget.Write( LogEventInfo ) method:

        protected override void Write(LogEventInfo logEvent)
        {
            string fileName = this.FileName.Render(logEvent);
            byte[] bytes = this.GetBytesToWrite(logEvent);

            if (this.ShouldAutoArchive(fileName, logEvent, bytes.Length))
            {
                this.InvalidateCacheItem(fileName);
                this.DoAutoArchive(fileName, logEvent);
                
    +           // FIX: The following two lines enable appending of header to each archived file.
    +           this.initializedFiles.Remove( fileName );
    +           this.initializedFilesCounter--;
            }

            this.WriteToFile(fileName, bytes, false);
        }

**Comment by hariniyedururi** [comment]
Has this issue been fixed? I see the changes made as suggested above in NLog FileTarget.cs but still have the issue if missing header. I tried both 3.2 and 4.0.1 versions.

**Comment by 304NotModified** [comment]
It should be fixed. I will check the code and the unit tests.

**Comment by 304NotModified** [comment]
The code of filetarget has been changed a lot after #425. but the unit test is still present and working.

See https://github.com/NLog/NLog/blob/master/tests/NLog.UnitTests/Targets/FileTargetTests.cs#l874

Which ArchiveNumbering do you use?

**Comment by hariniyedururi** [comment]
Hi Julian,

Here is the config

archiveAboveSize="1000000" archiveFileName="${basedir}/../logs/Metrics/archive/${event-context:item=ProcessName}.${event-context:item=HostSystemFileName}.${date:format=yyyyMMddHHmmssfff}.csv" maxArchiveFiles="100" fileName="${basedir}/../logs/Metrics/${event-context:item=ProcessName}.${event-context:item= HostSystemFileName }.csv" layout="${message}"

**Comment by 304NotModified** [comment]
So almost the same a the unit test. 

This is not full config? As I cannot see header property?

**Comment by hariniyedururi** [comment]
Please find the details in the attached mail.

**Comment by 304NotModified** [comment]
The header trick in you custom logger won't work. When we archive the file, we don't have any logEvents.

You can test it by giving the Header an initial value in the contructor of NLogCustomFileTarget . 

Closing this. If you need some help, please open a new ticket as this bug has been fixed.
